test/Graphics/Transform.cpp

```diff
@@ -0,0 +1,353 @@
+#include <SFML/Graphics/Transform.hpp>
+#include "GraphicsUtil.hpp"
+#include "SystemUtil.hpp"
+
+#include <doctest.h>
+
+using doctest::Approx;
+
+TEST_CASE("sf::Transform class - [graphics]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::Transform transform;
+            CHECK(transform == sf::Transform::Identity);
+            CHECK(transform.getMatrix()[0] == 1.0f);
+            CHECK(transform.getMatrix()[1] == 0.0f);
+            CHECK(transform.getMatrix()[2] == 0.0f);
+            CHECK(transform.getMatrix()[3] == 0.0f);
+            CHECK(transform.getMatrix()[4] == 0.0f);
+            CHECK(transform.getMatrix()[5] == 1.0f);
+            CHECK(transform.getMatrix()[6] == 0.0f);
+            CHECK(transform.getMatrix()[7] == 0.0f);
+            CHECK(transform.getMatrix()[8] == 0.0f);
+            CHECK(transform.getMatrix()[9] == 0.0f);
+            CHECK(transform.getMatrix()[10] == 1.0f);
+            CHECK(transform.getMatrix()[11] == 0.0f);
+            CHECK(transform.getMatrix()[12] == 0.0f);
+            CHECK(transform.getMatrix()[13] == 0.0f);
+            CHECK(transform.getMatrix()[14] == 0.0f);
+            CHECK(transform.getMatrix()[15] == 1.0f);
```

Comment:
I wonder if it's not better to compare the whole matrix at once. Sure, one would need to find out which elements differ, but with a good `operator<<` implementation, this should be quickly visible.

On the other hand, comparing each element means that a test will fail at the first possible element, not giving any context about the matrix's state and requiring debugging/manual print statements.

Maybe we could have something like
```cpp
std::array<float, 4 * 4> expected = {
   a, b, c, d,
   e, f, g, h,
   i, j, k, l
};
```
(with substituted values for the letters), which would also make the row/columns more clearer than a 1D view?

Reply:
>  comparing each element means that a test will fail at the first possible element

I don't believe this is true. At least with Catch2, `CHECK` will report the error and keep running unlike `REQUIRE` which aborts the test after the first failure. You're right though that comparing the whole matrix makes more sense. I was overusing `getMatrix()` when I started writing tests. I don't need to use that all over the place.

Reply:
You're right with `CHECK` vs. `REQUIRE`, remembered that wrong.

Printing both full matrices in case of a mismatch would still help, no?
Maybe this context makes it clearer _how_ the error happened.

Reply:
How else would we test the constructor? 
```cpp
CHECK(sf::Transform(10.0f, 11.0f, 12.0f, 13.0f, 14.0f, 15.0f, 16.0f, 17.0f, 18.0f)
   == sf::Transform(10.0f, 11.0f, 12.0f, 13.0f, 14.0f, 15.0f, 16.0f, 17.0f, 18.0f));
```
is a tautological comparison. It doesn't really hurt to include something like that but we need something else to compare the constructor against. Using `getMatrix()` to test the constructor is the equivalent of tests like this:
```cpp
SUBCASE("Default constructor")
{
    const sf::Vertex vertex;
    CHECK(vertex.position == sf::Vector2f(0.0f, 0.0f));
    CHECK(vertex.color == sf::Color(255, 255, 255));
    CHECK(vertex.texCoords == sf::Vector2f(0.0f, 0.0f));
}
```
Also notice how there is no explicit `SUBCASE("getMatrix()")` because that method is implicitly tested by this 3x3 constructor test as well as the `sf::Transform::Identity` test.

Reply:
> Printing both full matrices in case of a mismatch would still help, no?
 
We could try copy the `float*` return value from `getMatrix()` into a `std::array` then compare those two arrays which Doctest should automatically print to the console if a single element doesn't match. Does that sound good?

Reply:
Here's another option:
```cpp
            const sf::Transform transform(10.0f, 11.0f, 12.0f,
                                          13.0f, 14.0f, 15.0f,
                                          16.0f, 17.0f, 18.0f);
            const std::vector<float> matrix(transform.getMatrix(), transform.getMatrix() + 16);
            CHECK(matrix == std::vector<float>{10.0f, 13.0f, 0.0f, 16.0f,
                                               11.0f, 14.0f, 0.0f, 17.0f,
                                                0.0f,  0.0f, 1.0f,  0.0f,
                                               12.0f, 15.0f, 0.0f, 18.0f});
```
but Doctest won't automatically format this for us so I'd have to define another `operator<<` for this.